presto-main/src/main/java/com/facebook/presto/sql/analyzer/Scope.java

```diff
@@ -58,13 +60,15 @@ private Scope(
             boolean queryBoundary,
             RelationId relationId,
             RelationType relation,
-            Map<String, WithQuery> namedQueries)
+            Map<String, WithQuery> namedQueries,
+            Map<Identifier, WindowSpecification> windowSpecifications)
     {
         this.parent = requireNonNull(parent, "parent is null");
         this.relationId = requireNonNull(relationId, "relationId is null");
         this.queryBoundary = queryBoundary;
         this.relation = requireNonNull(relation, "relation is null");
         this.namedQueries = ImmutableMap.copyOf(requireNonNull(namedQueries, "namedQueries is null"));
+        this.windowSpecifications = ImmutableMap.copyOf(requireNonNull(windowSpecifications, "windowSpecifications is null"));
```

Comment:
Does it? Can you please give a reference? I couldn't find it here: https://google.github.io/guava/releases/21.0/api/docs/com/google/common/collect/ImmutableMap.html#copyOf-java.util.Map- (it only says about throwing an exception if any key or value is null. Doesn't say anything about what happens if the map argument is null). Also, I found plenty of such constructs in Presto codebase.